File: app/src/main/java/me/saket/dank/utils/CommentSortUtils.java
package me.saket.dank.utils;

import android.support.annotation.StringRes;

import net.dean.jraw.models.CommentSort;

import me.saket.dank.R;

// TODO Kotlin: Convert to extension functions.
public class CommentSortUtils {

  @StringRes
  public static int sortingDisplayTextRes(CommentSort sort) {
    switch (sort) {
      case CONFIDENCE:
        return R.string.comment_sorting_best;

      case TOP:
        return R.string.comment_sorting_top;

      case NEW:
        return R.string.comment_sorting_new;

      case HOT:
        throw new AssertionError("HOT comment sort isn't supported anymore");

      case CONTROVERSIAL:
        return R.string.comment_sorting_controversial;

      case OLD:
        return R.string.comment_sorting_old;

      case QA:
        return R.string.comment_sorting_qna;

      default:
        throw new UnsupportedOperationException("Unknown comment sort: " + sort);
    }
  }
}

File: app/src/main/java/me/saket/dank/utils/CommentSortUtils.kt
package me.saket.dank.utils

import androidx.annotation.StringRes
import me.saket.dank.R
import net.dean.jraw.models.CommentSort
import net.dean.jraw.models.CommentSort.*

// TODO Kotlin: Convert to extension functions?
object CommentSortUtils {

  @StringRes
  fun sortingDisplayTextRes(sort: CommentSort): Int {
    return when (sort) {
      CONFIDENCE -> R.string.comment_sorting_best
      TOP -> R.string.comment_sorting_top
      NEW -> R.string.comment_sorting_new
      CONTROVERSIAL -> R.string.comment_sorting_controversial
      OLD -> R.string.comment_sorting_old
      QA -> R.string.comment_sorting_qna
      RANDOM -> R.string.comment_sorting_random
      LIVE -> R.string.comment_sorting_live
    }
  }
}
